code/modules/mob/living/carbon/xenomorph/castes/hivemind/abilities_hivemind.dm

```diff
@@ -1 +1,208 @@
+	owner.playsound_local(owner.loc, 'sound/voice/alien_drool1.ogg', 50, TRUE)
+	to_chat(owner, "<span class='xenonotice'>We regain the strength to reposition our neural core.</span>")
+	return ..()
+
+/datum/action/xeno_action/activable/reposition_core/can_use_ability(atom/target, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+	if(QDELETED(target))
+		return FALSE
+
+	var/turf/target_turf = target
+	var/mob/living/carbon/xenomorph/hivemind/X = owner
+
+	if(!istype(X))
+		return FALSE
+
+	if(!check_build_location(target_turf, X))
+		return FALSE
+
+/datum/action/xeno_action/activable/reposition_core/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/hivemind/X = owner
+	var/obj/effect/alien/hivemindcore/core = X.core
+	var/turf/target_turf = get_turf(X)
+	var/distance = get_dist(target_turf, get_turf(core))
+
+	var/delay = max(HIVEMIND_REPOSITION_CORE_DELAY_MIN, HIVEMIND_REPOSITION_CORE_DELAY_MOD * distance) //Calculate the distance scaling delay before we complete the reposition
+
+
+	if(!check_build_location(target_turf, X)) //Check target turf for suitability
+		return fail_activate()
+	to_chat(owner, "<span class='xenodanger'>We begin the process of transfering our consciousness... We estimate this will require [delay * 0.1] seconds.</span>")
+
+	succeed_activate()
+	if(!do_after(X, delay, TRUE, null, BUSY_ICON_BUILD))
+		if(!QDELETED(src))
+			to_chat(X, "<span class='xenodanger'>We abort transferring our consciousness, expending our precious plasma for naught.</span>")
+			return fail_activate()
+
+	if(!can_use_ability(target_turf, FALSE, XACT_IGNORE_PLASMA))
+		return fail_activate()
+
+	core.forceMove(get_turf(target_turf)) //Move the core
+	playsound(core.loc, "alien_resin_build", 50)
+	core.obj_integrity = 1 //Reset the core's health to 1; travelling is hard, exhausting work!
+	to_chat(X, "<span class='xenodanger'>We succeed in transferring our consciousness to a new neural core!</span>")
+
+	add_cooldown()
+
+	GLOB.round_statistics.hivemind_reposition_core_uses++ //Increment the statistics
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "hivemind_reposition_core_uses")
+
+	X.hivemind_core_alert() //Alert the hive and marines
+
+/datum/action/xeno_action/activable/reposition_core/proc/check_build_location(turf/target_turf, mob/living/carbon/xenomorph/hivemind/X)
+
+	if(!X || !target_turf) //Sanity
+		return FALSE
+
+	if(target_turf.z != X.core.z)
+		to_chat(X, "<span class='xenodanger'>We cannot transfer our core here.</span>")
+		return FALSE
+
+	if(target_turf.density) //Check to see if there's room
+		to_chat(X, "<span class='xenodanger'>We require adequate room to transfer our core.</span>")
+		return FALSE
+
+	var/obj/effect/alien/weeds/W = locate() in range(0, target_turf) //Make sure we actually have weeds at our destination.
+	if(!W)
+		to_chat(X, "<span class='xenodanger'>There are no weeds for us to transfer our consciousness to!</span>")
+		return FALSE
```

Comment:
Just have this as a
 if(locate(/alien/weeds) in target_turf

Reply:
Done.